README.md

```diff
@@ -62,7 +62,7 @@ Shomei can be launched in two different modes:
 ### Mode 1: Trace Generation
 
 The first mode is used for the generation of traces so that the prover can retrieve them.
-To use this mode, you need to activate it with these flags: `--enable-trace-generation=true` and `--trace-start-block-number=BLOCK_NUMBER`. The `--trace-start-block-number` flag allows you to define from when we want to start the generation of traces.
+To use this mode, you need to activate it with these flags: `--enable-trace-generation=true` and `--trace-start-block-number=BLOCK_NUMBER`. The `--trace-start-block-number` flag allows you to define on when we want to start the generation of traces.
```

Comment:
"define on when" ? native english speaker here, this is not an improvement.  I think the original text is sufficient but if we are going for grammatical correctness I would suggest 
```suggestion
To use this mode, you need to activate it with these flags: `--enable-trace-generation=true` and `--trace-start-block-number=BLOCK_NUMBER`. The `--trace-start-block-number` flag allows you to define at which block number we want to start the generation of traces.
```